### PR TITLE
feat: Add gutter controls to keyboard accessibility mode

### PR DESCRIPTION
**Pull request by akoreman**
*Issue: #5125,  #5145*

Adds operating the fold controls and annotations in the gutter to the keyboard accessibility mode. When tabbing through the page, you can focus on the gutter div. When pressing enter, you enter the gutter and can use the arrow keys to navigate through the controls in the gutter. Press enter to interact with the controls. Press escape to set focus back to the gutter div and be able to TAB through the page again.

Only iterates over gutter controls in current viewbox to prevent potentially scrolling the cursor outside of the current view (which can cause a11y problems).

https://user-images.githubusercontent.com/34573235/235914151-b3c64385-57ba-4f5a-9237-633042301037.mov

Additionally adds title attribute to folding buttons:

Screenshot 2023-05-01 200955

Screenshot 2023-05-01 201017

Changes to `src/mouse/default_gutter_handler.js` are moving things around to to it easier to use `GutterTooltip` outside of `GutterMouseHandler`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/editor.js [hunk 1]
```diff
@@ -2874,34 +2875,68 @@ config.defineOptions(Editor.prototype, "editor", {
...
+                this.renderer.content.setAttribute("aria-roledescription", "");
+                this.renderer.content.classList.remove(this.keyboardFocusClassName);
                 this.renderer.content.setAttribute("aria-label", "");
             
                 this.renderer.content.removeEventListener("keyup", focusOnEnterKeyup.bind(this));
                 this.commands.removeCommand(blurCommand);
+
+                this.renderer.$gutter.setAttribute("tabindex", -1);
+                this.renderer.$gutter.setAttribute("aria-hidden", true);
+                this.renderer.content.setAttribute("role", "");
```
Comment (oykuyilmaz) [inline src/editor.js:2933]: What is a best approach for removing the aria attributes, is it set them to empty value or remove them completely. I guess I followed a different approach and asking to be consistent.

File: src/keyboard/textinput.js [hunk 2]
```diff
@@ -61,9 +61,21 @@ var TextInput = function(parentNode, host) {
         }
         if (options.role) {
             text.setAttribute("role", options.role);
-        }
+        }     
+    };
+    this.setAriaLabel = function() {
```
Comment (oykuyilmaz) [inline src/keyboard/textinput.js:66]: can we maybe merge two functions here: `setAriaOptions` and `setAriaLabel`? They seem serving similar purposes.

File: src/editor.js (hunk 1 above)
Comment (akoreman) [reply src/editor.js:2933]: I don't think there are differences between how assistive tech interpret it but it probably makes more sense to just remove the attributes. I'll update the PR.

File: src/keyboard/textinput.js (hunk 2 above)
Comment (akoreman) [reply src/keyboard/textinput.js:66]: If we merge the functions, we'd need to make sure that whenever we call `setAriaLabel` we pass the same `options` as the last time we called `setAriaOptions` which I think would be a hassle given that e.g. autocomplete extensions also call this function.

File: src/keyboard/gutter_handler.js [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+                }
+
+                if (nearestFoldIndex !== null && nearestAnnotationIndex === null){
+                    this.activeRowIndex = nearestFoldIndex;
+                    this.activeLane = "fold";
+                    this.$focusFoldWidget(this.activeRowIndex);
+                    return;
+                }
+
+                if (Math.abs(nearestAnnotationIndex - index) < Math.abs(nearestFoldIndex - index)){
```
Comment (oykuyilmaz) [inline src/keyboard/gutter_handler.js:73]: if they are the same, `nearestFoldIndex` will get the keyboard focus, isn't it?

**Review by oykuyilmaz — APPROVED** [review]

**Pull request merged**